Step1 init commit

derren-korean opened the pull request:
점수 계산을 제외한 볼링 게임 점수판을 구현.

derren-korean committed:
init commit : 점수객체 구현

derren-korean committed:
플레이어 기본 내역 출력까지 구현

derren-korean committed:
숫자 범위 제한 및 사용자 입력 모드 전단계

derren-korean committed:
수동으로 플레이해보기, 점수계산 이전까지 구현

javajigi commented on src/main/java/bowling/domain/Name.java:
```diff
@@ -0,0 +1,18 @@
+
+    public Name(String name) {
+        this.name = name;
+        if (!InputView.isName(name)) throw new IllegalArgumentException();
```
InputView에 의존관계를 가지는 것이 찜찜하네요.

javajigi commented on src/test/java/bowling/domain/RollTest.java:
```diff
@@ -0,0 +1,49 @@
+        assertEquals("7", new Roll(new Pins(7)).toBowlingScore().toString());
+        assertEquals("8", new Roll(new Pins(8)).toBowlingScore().toString());
+        assertEquals("9", new Roll(new Pins(9)).toBowlingScore().toString());
+        assertEquals("/", new Roll(new Pins(10)).toBowlingScore().toString());
```
이런 건 중복이 너무 많은데요. 반복문 통해 테스트 코드 검증해도 좋을 것 같아요.

javajigi commented on src/main/java/bowling/domain/Score/Roll.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.Objects;
+import java.util.Random;
+
+public class Roll {
```
굳이 Roll과 Pins를 분리할 필요가 있을까요?
둘을 하나로 합치는 것이 좀 더 객체 구조가 단순해 지지 않을까요?
둘을 분리해야 하는 이유는?

javajigi commented on src/main/java/bowling/domain/frame/Frame.java:
```diff
@@ -0,0 +1,49 @@
+        this.score = score;
+    }
+
+    public Frame(Frame frame) {
```
Frame보다는 `public Frame(Score score, TotalScore totalScore)`로 구현하는 것은 어떨까요?
앞에 Score를 인자로 받는 것과 고려해 봤을 때요..

javajigi commented on src/main/java/bowling/domain/number/BowlingScore.java:
```diff
@@ -0,0 +1,57 @@
+                return String.valueOf(this.ordinal());
+            default:
+                return "";
+        }
```
이 메소드 뭔가 마음에 들지 않는데요.
이렇게 구현할 바에 enum 생성할 때 `ONE(""), SPARE("/")`와 같이 생성하는 것이 낫지 않을까요?

javajigi reviewed (changes requested):
전체적으로 원시 타입을 래핑하면서 코드도 깔끔하게 구현했네요. 그런데 원시 타입을 래핑한 객체들이 유효성 체크말고 다른 로직들도 구현함으로써 객체의 수를 좀 줄이는 시도를 해보면 좋을 것 같아요.

derren-korean committed:
피드백 반영 :
Name.java : io와 의존관계 제거,
RollTest.java : 테스트 코드 반복문 적용,
Roll.java : Pins제거하여 불필요한 객체 분리.(답변 및 질문은 offline에서 할께요~)
BoewlingScore.java : enum생성자를 이용하여 코드 간결화.
Frame 및 Score Package는 해당 변수로 생성자 만들도록 변경. (ex : public Frame(Frame frame){...} -> public Frame(Score score, TotalScore totalScore)
